openmc/data/resonance_covariance.py

```diff
@@ -0,0 +1,743 @@
+    energy_min : float
+        Minimum energy of the resolved resonance range in eV
+    energy_max : float
+        Maximum energy of the resolved resonance range in eV
+
+    Attributes
+    ----------
+    energy_min : float
+        Minimum energy of the resolved resonance range in eV
+    energy_max : float
+        Maximum energy of the resolved resonance range in eV
+    parameters : pandas.DataFrame
+        Resonance parameters
+    covariance : numpy.array
+        The covariance matrix contained within the ENDF evaluation
+    lcomp : int
+        Flag indicating format of the covariance matrix within the ENDF file
+    mpar : int
+        Number of parameters in covariance matrix for each individual resonance
+    formalism : str
+        String descriptor of formalism
+    """
+    def __init__(self, energy_min, energy_max):
+        self.energy_min = energy_min
+        self.energy_max = energy_max
+
+    def res_subset(self, parameter_str, bounds):
+        """Produce a subset of resonance parameters and the corresponding
+        covariance matrix to an IncidentNeutron object.
+
+        Parameters
+        ----------
+        parameter_str : str
+            parameter to be discriminated
+            (i.e. 'energy', 'captureWidth', 'fissionWidthA'...)
+        bounds : np.array
+            [low numerical bound, high numerical bound]
+
+        Returns
+        -------
+        res_cov_range : openmc.data.ResonanceCovarianceRange
+            ResonanceCovarianceRange object that contains a subset of the
+            covariance matrix (upper triangular) as well as a subset parameters
+            within self.file2params
+
+        """
+        # Copy range and prevent change of original
+        res_cov_range = copy.deepcopy(self)
+
+        parameters = self.file2res.parameters
+        cov = res_cov_range.covariance
+        mpar = res_cov_range.mpar
+        # Create mask
+        mask1 = parameters[parameter_str] >= bounds[0]
+        mask2 = parameters[parameter_str] <= bounds[1]
+        mask = mask1 & mask2
+        res_cov_range.parameters = parameters[mask]
+        indices = res_cov_range.parameters.index.values
+        # Build subset of covariance
+        sub_cov_dim = len(indices)*mpar
+        cov_subset_vals = []
+        for index1 in indices:
+            for i in range(mpar):
+                for index2 in indices:
+                    for j in range(mpar):
+                        if index2*mpar+j >= index1*mpar+i:
+                            cov_subset_vals.append(cov[index1*mpar+i,
+                                                   index2*mpar+j])
+
+        cov_subset = np.zeros([sub_cov_dim, sub_cov_dim])
+        tri_indices = np.triu_indices(sub_cov_dim)
+        cov_subset[tri_indices] = cov_subset_vals
+
+        res_cov_range.file2res.parameters = parameters[mask]
+        res_cov_range.covariance = cov_subset
+        return res_cov_range
+
+    def sample_resonance_parameters(self, n_samples):
```

Comment:
What do you think about naming this simply `sample()`?

Reply:
Makes sense to me. Wasn't sure how descriptive the names needed to be but as with `res_subset` I think that having them as methods of the objects provides enough context